database/scripts/MetadataImport.sql

```diff
@@ -1,175 +1,233 @@
+                mappedColumns = array_append(mappedColumns, foundTargetColumn);
+                typedColumns = array_append(typedColumns, foundTargetColumn || (CASE WHEN foundTargetColumn = targetBandnumberColumn THEN ' integer' ELSE ' text' END));
+            END IF;
+        END LOOP;
+
+        -- check mandatory columns
+
+    IF NOT targetBandnumberColumn = ANY(mappedColumns) THEN
+        SELECT csvColumn FROM metaColumnsMap WHERE targetColumn = targetTitleColumn INTO csvTitleColumn;
+        RAISE NOTICE E'The mandatory Bandnumber column ("%") is missing from the input file.\n'
+                       'Check the file and the column mapping.\n'
+                       'The procedure has been aborted.', csvTitleColumn;
+                EXIT importMetaScript;
+    END IF;
+
+    IF NOT targetCategoryColumn = ANY(mappedColumns) THEN
+        SELECT csvColumn FROM metaColumnsMap WHERE targetColumn = targetCategoryColumn INTO csvTitleColumn;
+        RAISE NOTICE E'The mandatory Category column ("%") is missing from the input file.\n'
+                      'Check the file and the column mapping.\n'
+                      'Note that ''Category'' should say either "Ecosystem" or "Pressure" \n'
+                      'The procedure has been aborted.', csvTitleColumn;
+                EXIT importMetaScript;
+    END IF;
+
+    IF NOT targetLanguageColumn = ANY(mappedColumns) THEN
+    SELECT csvColumn FROM metaColumnsMap WHERE targetColumn = targetLanguageColumn INTO csvTitleColumn;
+    RAISE NOTICE E'The mandatory Language column ("%") is missing from the input file.\n'
+                   'Check the file and the column mapping.\n'
+                   'The procedure has been aborted.', csvTitleColumn;
+            EXIT importMetaScript;
+    END IF;
+
+    IF NOT targetTitleColumn = ANY(mappedColumns) THEN
+        SELECT csvColumn FROM metaColumnsMap WHERE targetColumn = targetTitleColumn INTO csvTitleColumn;
+        RAISE NOTICE E'A component title column ("%") is missing from the input file.\n'
+                      'Check the file and the column mapping.\n'
+                      'The procedure has been aborted.', csvTitleColumn;
+                EXIT importMetaScript;
+    END IF;
+
+    IF faultyCol > 0 THEN
+        RAISE WARNING E'% unrecognised column% in the input file: %.\n'
+                       'The procedure has been aborted.',
+                       faultyCol, CASE WHEN faultyCol = 1 THEN '' ELSE 's' END,
+                       array_to_string(unmappedColumns, ', ');
+    ELSE
+        formatcmd = format(
+            (CASE WHEN winOS THEN
+             '''chcp 65001 && powershell "Import-Csv """%1$s""" -Delimiter """;""" | Select-Object %2$s | '
+                                         'ConvertTo-Csv -NoTypeInformation -Delimiter """;""" | Select-Object -Skip 1"'''
+                 ELSE
+                 '''cut -d ";" -f %3$s %1$s''' END),
```

Comment:
Unix version is overly simplistic and works only for very "well behaved" data.
In particular, quoted fields will get corrupted. Might be possible to mend using awk.